cmd/gnodev/precompile.go

```diff
@@ -108,15 +125,27 @@ func precompileFile(srcPath string, opts precompileOptions) error {
 	}
 
 	// preprocess.
-	transformed, err := gno.Precompile(string(source), tags, srcPath)
+	f, transformed, err := gno.Precompile(string(source), tags, srcPath)
 	if err != nil {
 		return fmt.Errorf("%w", err)
 	}
 
+	// get imported package path
+	var importPkgPaths []string
+	for _, i := range f.Imports {
+		importPath := i.Path.Value[1 : len(i.Path.Value)-1]
+		prefix := "github.com/gnolang/gno"
+		if strings.HasPrefix(importPath, prefix) {
+			res := strings.TrimPrefix(importPath, prefix)
+			importPkgPaths = append(importPkgPaths, "."+res)
+		}
+	}
```

Comment:
What do you think about extracting this out into a separate method, like `extractImportPaths`?
You also don't need to redefine the prefix on each loop iteration.

ex
```go
const (
	packagePrefix = "github.com/gnolang/gno"
)

// extractImportPaths returns the extracted import paths
// from the file that match the gno packagePrefix
func extractImportPaths(file *ast.File) []string {
	importPaths := make([]string, 0, len(file.Imports))

	for _, i := range file.Imports {
		importPath := i.Path.Value[1 : len(i.Path.Value)-1]

		if strings.HasPrefix(importPath, packagePrefix) {
			importPaths = append(
				importPaths,
				"."+strings.TrimPrefix(importPath, packagePrefix),
			)
		}
	}

	return importPaths
}
```

This allows you to easily write a unit test for this method and validate it works correctly.

Reply:
Resolving this convo as you've rewritten the implementation